New package: shotgun-2.5.0

realcharmer opened the pull request:
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures:
  - aarch64-musl
  - aarch64
  - armv6l-musl
  - armv6l
  - armv7l-musl
  - armv7l
  - x86_64-musl
  - x86_64

This package is based on #20555 and closes #20282.

paper42 commented on srcpkgs/shotgun/template:
```diff
@@ -0,0 +1,24 @@
+# Template file for 'shotgun'
+pkgname=shotgun
+version=2.2.1
```
2.3.1 seems to be the latest one upstream

realcharmer replied:
It has been bumped to 2.3.1 in master, but there has been no release with this version yet. It might be a version number mismatch, but this package works with the release version. See https://github.com/neXromancers/shotgun/releases.

paper42 replied:
I think they just do tags and not releases now. Most other distributions package 2.3.1, so we should do so too: https://repology.org/project/shotgun/versions

realcharmer replied:
Ok, I'll update the package version.

realcharmer replied:
Should be ready.

paper42 replied:
commit message should also be adjusted

realcharmer replied:
I forgot about that ... fixed.

paper42 commented on srcpkgs/shotgun/template:
```diff
@@ -0,0 +1,24 @@
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="libX11-devel libXrandr-devel"
+depends="libX11 libXrandr"
```
```suggestion
```
all of these dependencies are autodetected

realcharmer commented:
Bump. Still waiting for merge.

realcharmer commented:
Updated to version `2.5.0` and rebased to current `master`. This new release is now ported to `x11rb` and doesn't depend on any external libraries, so we can drop `libX11-devel` and `libXrandr-devel` from `makedepends`.
